src/connectors/refinement-list/connectRefinementList.js

```diff
@@ -374,11 +388,9 @@ export default function connectRefinementList(renderFn, unmountFn = noop) {
         unmountFn();
 
         if (operator === 'and') {
-          return state.removeFacetRefinement(attribute).removeFacet(attribute);
+          return state.removeFacet(attribute);
         } else {
-          return state
-            .removeDisjunctiveFacetRefinement(attribute)
-            .removeDisjunctiveFacet(attribute);
+          return state.removeDisjunctiveFacet(attribute);
```

Comment:
We can cover this by a test that ensure that the value is not there anymore (`dispose` is never tested).

Reply:
7414f92,  noticed that we weren't removing the maxValuesPerFacet too